common/util.go

```diff
@@ -410,3 +411,45 @@ func ReadAtMost(r io.Reader, n int) ([]byte, error) {
 	}
 	return b, err
 }
+
+func enumerateNvidiaDevices() []string {
+	gpu, err := ghw.GPU()
+	if err != nil {
+		glog.Errorf("Error getting GPU info: %v", err)
+		return nil
+	}
+
+	nvidiaCardCount := 0
+	for _, card := range gpu.GraphicsCards {
+		if strings.EqualFold(card.DeviceInfo.Vendor.Name[:6], "nvidia") {
+			nvidiaCardCount += 1
+		}
+	}
+	if nvidiaCardCount == 0 {
+		glog.Warning("No nvidia devices found")
+		return nil
+	}
+
+	devices := []string{}
+
+	for i := 0; i < nvidiaCardCount; i++ {
+		s := strconv.Itoa(i)
+		devices = append(devices, s)
+	}
+
+	return devices
+}
+
+func ParseNvidiaDevicesFlag(nvidia string) []string {
+	var devices []string
+	if nvidia == "all" {
+		devices = enumerateNvidiaDevices()
+		if devices == nil || len(devices) == 0 {
```

Comment:
Devices can never be `nil` as it's a value type (all arrays in golang are value types), we can shorten this to `len(devices) == 0`

See https://play.golang.org/p/evGL_j96MHb

```go
package main

import (
	"fmt"
)

func main() {
 	a :=[]string{}
	b := []*string{}
	a = nil
	b = nil
	fmt.Println(a, b)
}
```

Console output: 
```bash
[] []

```

Reply:
Ah TIL, will update!